app/assets/javascripts/admin/addon/components/admin-config-areas/webhooks-form.gjs

```diff
@@ -0,0 +1,301 @@
+import { cached, tracked } from "@glimmer/tracking";
+import { concat, hash } from "@ember/helper";
+import { action } from "@ember/object";
+import { service } from "@ember/service";
+import { eq } from "truth-helpers";
+import BackButton from "discourse/components/back-button";
+import ConditionalLoadingSection from "discourse/components/conditional-loading-section";
+import Form from "discourse/components/form";
+import GroupSelector from "discourse/components/group-selector";
+import PluginOutlet from "discourse/components/plugin-outlet";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import { i18n } from "discourse-i18n";
+import WebhookEventChooser from "admin/components/webhook-event-chooser";
+import CategorySelector from "select-kit/components/category-selector";
+import TagChooser from "select-kit/components/tag-chooser";
+
+export default class AdminConfigAreasWebhookForm extends Component {
+  @service router;
+  @service siteSettings;
+  @service store;
+
+  @tracked webhook = this.args.webhook;
+
+  @tracked loadingExtras = true;
+
+  @tracked webhookEventTypes = [];
+  @tracked defaultEventTypes = {};
+  @tracked groupedEventTypes = {};
+  @tracked contentTypes = [];
+  @tracked deliveryStatuses = [];
+
+  constructor() {
+    super(...arguments);
+
+    this.#loadExtras();
+  }
+
+  @cached
+  get formData() {
+    return {
+      payload_url: this.webhook.payload_url,
+      content_type: this.webhook.content_type,
+      secret: this.webhook.secret === "" ? null : this.webhook.secret,
+      categories: this.webhook.categories,
+      group_names: this.webhook.group_names,
+      tag_names: this.webhook.tag_names,
+      wildcard: this.webhook.wildcard,
+      web_hook_event_types: this.webhook.web_hook_event_types,
+      verify_certificate: this.webhook.verify_certificate,
+      active: this.webhook.active,
+    };
+  }
+
+  async #loadExtras() {
+    try {
+      this.loadingExtras = true;
+
+      const webhooks = await this.store.findAll("web-hook");
+
+      this.groupedEventTypes = webhooks.extras.grouped_event_types;
+      this.defaultEventTypes = webhooks.extras.default_event_types;
+      this.contentTypes = webhooks.extras.content_types;
+      this.deliveryStatuses = webhooks.extras.delivery_statuses;
+
+      if (this.webhook.isNew) {
+        this.webhookEventTypes = [...this.defaultEventTypes];
+      } else {
+        this.webhookEventTypes = [...this.webhook.web_hook_event_types];
+      }
+    } finally {
+      this.loadingExtras = false;
+    }
+  }
+
+  get showTagsFilter() {
+    return this.siteSettings.tagging_enabled;
+  }
+
+  get saveButtonLabel() {
+    return this.webhook.isNew
+      ? "admin.web_hooks.create"
+      : "admin.web_hooks.save";
+  }
+
+  @action
+  async save(data) {
+    try {
+      const isNew = this.webhook.isNew;
+
+      this.webhook.setProperties({
+        ...data,
+        web_hook_event_types: this.webhookEventTypes,
```

Comment:
Didn't convert these checkboxes into FormKit as part of this PR. Can be done as a follow-up.